### ISSUE TITLE
EVP_MAC API should not allow update calls after final call

### ISSUE THREAD

**Issue by kroeckx** [openssl/openssl#13286]
This is a split of #13210, for just the EVP_MAC layer, which I think we should fix before 3.0.

**Comment by nhorman** [comment]
@kroeckx can you elaborate on what needs to be done in this task?

**Comment by nhorman** [comment]
no response, marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by paulidale** [comment]
Would it be worthwhile having a single issue about enforcing the documented algorithm life cycles (originally introduced by #14522)?

**Comment by nhorman** [comment]
@paulidale in principle that sounds reasonable.  I'm not quite sure where that might fit on the development schedule though.  Do you think such a task would make sense as a community issue (or issues)?

**Comment by paulidale** [comment]
I've no idea.  I'm not sure if the community would take it up or not.
I'd like to take a stab at it but am on other things for the foreseeable future and then it would be very low priority I suspect.

**Comment by nhorman** [comment]
yeah, thats the issue, no one has time for stuff like this.  I'm going to mark this as a community issue for now, just so it stays open.  If someone gets to it, great, if not, we can revisit during a future planning session

**Comment by t8m** [comment]
This could be done in similar way as we did for digests:
https://github.com/openssl/openssl/pull/20375/files
